openai-model-provider/openaiproxy/proxy.go

```diff
@@ -70,18 +70,17 @@ func (s *Server) Openaiv1ProxyRedirect(req *http.Request) {
 	}
 }
 
-func modifyRequestBodyForO1(reqBody *openai.ChatCompletionRequest) {
-	reqBody.Temperature = nil
+func modifyRequestBodyForReasoningModel(reqBody *openai.ChatCompletionRequest) {
+	temp := float32(1.0) // temperature is not supported for reasoning models. It must be 1.0
+	reqBody.Temperature = &temp
 	for i, msg := range reqBody.Messages {
 		if msg.Role == "system" {
 			reqBody.Messages[i].Role = "developer"
 		}
 	}
 }
 
-func isModelO1(model string) bool {
-	if model == "o1" {
-		return true
-	}
-	return strings.HasPrefix(model, "o1-") && !strings.HasPrefix(model, "o1-mini") && !strings.HasPrefix(model, "o1-preview")
+func isReasoningModel(model string) bool {
+	model = strings.TrimSpace(model)
```

Comment:
This shouldn't be necessary. I would guess that any extraneous space would mess up a bunch of other things.

Reply:
fair.